tests/integration-tests/tests/test_fsx_lustre.py

```diff
@@ -0,0 +1,116 @@
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    _test_fsx_lustre_correctly_mounted(remote_command_executor, mount_dir)
+    _test_import_path(remote_command_executor, mount_dir)
+    _test_fsx_lustre_correctly_shared(remote_command_executor, mount_dir)
+    _test_export_path(remote_command_executor, mount_dir, bucket_name)
+
+
+def _test_fsx_lustre_correctly_mounted(remote_command_executor, mount_dir):
+    logging.info("Testing fsx lustre is correctly mounted")
+    result = remote_command_executor.run_remote_command("df -h -t lustre --output=source,size,target | tail -n +2")
+    assert_that(result.stdout).matches(r"[0-9\.]+@tcp:/fsx\s+3\.4T\s+{mount_dir}".format(mount_dir=mount_dir))
+
+    result = remote_command_executor.run_remote_command("cat /etc/fstab")
+    assert_that(result.stdout).matches(
+        r"fs-[0-9a-z]+\.fsx\.[a-z1-9\-]+\.amazonaws\.com:/fsx {mount_dir} lustre defaults,_netdev 0 0".format(
+            mount_dir=mount_dir
+        )
+    )
+
+
+def _test_import_path(remote_command_executor, mount_dir):
+    logging.info("Testing fsx lustre import path")
+    result = remote_command_executor.run_remote_command("cat {mount_dir}/s3_test_file".format(mount_dir=mount_dir))
+    assert_that(result.stdout).is_equal_to("Downloaded by FSx Lustre")
+
+
+def _test_fsx_lustre_correctly_shared(remote_command_executor, mount_dir):
+    logging.info("Testing fsx lustre correctly mounted on compute nodes")
+    remote_command_executor.run_remote_command("touch {mount_dir}/test_file".format(mount_dir=mount_dir))
+    job_command = (
+        "cat {mount_dir}/s3_test_file "
+        "&& cat {mount_dir}/test_file "
+        "&& touch {mount_dir}/compute_output".format(mount_dir=mount_dir)
+    )
+    result = remote_command_executor.run_remote_command("echo '{0}' | qsub".format(job_command))
+    job_id = _assert_job_submitted(result.stdout)
+    _wait_job_completed(remote_command_executor, job_id)
+    status = _get_job_exit_status(remote_command_executor, job_id)
+    assert_that(status).is_equal_to("0")
+    remote_command_executor.run_remote_command("cat {mount_dir}/compute_output".format(mount_dir=mount_dir))
+
+
+def _test_export_path(remote_command_executor, mount_dir, bucket_name):
+    logging.info("Testing fsx lustre export path")
+    remote_command_executor.run_remote_command(
+        "echo 'Exported by FSx Lustre' > {mount_dir}/file_to_export".format(mount_dir=mount_dir)
+    )
+    remote_command_executor.run_remote_command(
+        "sudo lfs hsm_archive {mount_dir}/file_to_export && sleep 5".format(mount_dir=mount_dir)
+    )
+    remote_command_executor.run_remote_command(
+        "aws s3 cp s3://{bucket_name}/export_dir/file_to_export ./file_to_export".format(bucket_name=bucket_name)
+    )
+    result = remote_command_executor.run_remote_command("cat ./file_to_export")
+    assert_that(result.stdout).is_equal_to("Exported by FSx Lustre")
+
+
+def _assert_job_submitted(qsub_output):
```

Comment:
I've already seen this function in the awsbatch test. We should move this in the a utils module.

Reply:
these functions are specific for the type of scheduler used (sge here, awsbatch in the other file). I was planning to move them to a common module only when actually multiple tests call them.